core/src/player.rs

```diff
@@ -712,23 +708,54 @@ impl Player {
         callback: Object<'gc>,
         context: &mut UpdateContext<'_, 'gc>,
     ) {
-        if let Some(root_clip) = context.stage.root_clip() {
-            let mut activation = Activation::from_nothing(
-                context.reborrow(),
-                ActivationIdentifier::root("[Context Menu Callback]"),
-                root_clip,
-            );
+        if let Some(menu_state) = context.current_context_menu {
+            if let Some(display_object) = menu_state.get_display_object() {
+                let mut activation = Activation::from_nothing(
+                    context.reborrow(),
+                    ActivationIdentifier::root("[Context Menu Callback]"),
+                    display_object,
+                );
 
-            // TODO: Remember to also change the first arg
-            // when we support contextmenu on non-root-movie
-            let params = vec![root_clip.object(), Value::Object(item)];
+                let params = vec![display_object.object(), Value::Object(item)];
 
-            let _ = callback.call(
-                "[Context Menu Callback]".into(),
-                &mut activation,
-                Value::Undefined,
-                &params,
-            );
+                let _ = callback.call(
+                    "[Context Menu Callback]".into(),
+                    &mut activation,
+                    Value::Undefined,
+                    &params,
+                );
+            }
+        }
+    }
+
+    ///Returns the first display object that the mouse is hovering over that has a custom context menu. Returns root if none is found.
+    fn get_context_menu_display_object<'gc>(
+        context: &mut UpdateContext<'_, 'gc>,
+    ) -> Option<DisplayObject<'gc>> {
+        let mut display_object = run_mouse_pick(context, false).map(|obj| obj.as_displayobject());
+        loop {
+            if let Some(disp_obj) = display_object {
+                if disp_obj.hit_test_shape(
```

Comment:
wait, why is this hit test needed? Can the reason be documented? Or if you've found an edge case, let me know and I can try make a test.
Intuitively, if we've found a hit via `run_mouse_pick`, we shouldn't need to test for hit tests again while recursing upward.

Reply:
I don't exactly know what I thought 5 months ago, but I think I did indeed mess up.
I can't find a reason for this test after testing with my test swf I made back then.
I do remember using the flash player that came with macromedia flash 8 back then which did not give the same results using the latest flash player. So I might have taught this was needed because of inconsistencies with the old flash player I was using and the newest one.
Nevertheless, this does indeed not seem necessary.